src/types/auth.ts

```diff
@@ -0,0 +1,25 @@
+import { Session } from 'next-auth'
+
+/**
+ * Object representing an authentication session.
+ */
+type SessionData = Omit<Session, 'user'> & {
+	id?: 'string'
+}
+
+/**
+ * Object representing an authenticated user.
+ */
+type UserData = Session['user'] & {
+	nickname?: string | null
+}
+
+/**
+ * Enumeration of supported auth provider IDs.
+ */
+const ValidAuthProviderId = {
+	CloudIdp: 'cloud-idp',
+} as const
```

Comment:
I had that at first, but I can't remember why I switched away from it. 🤔 I switched between multiple approaches because some weren't' working the way I expected. But I'll double-check and see if that works with everything else!

Reply:
Updated! ✨ https://github.com/hashicorp/dev-portal/pull/728/commits/29ffbaee3131c187e705493ec5a65f2accf6a698